MoreLinq/Pad.cs

```diff
@@ -111,11 +111,11 @@ public static IEnumerable<TSource> Pad<TSource>(this IEnumerable<TSource> source
             if (source == null) throw new ArgumentNullException(nameof(source));
             if (paddingSelector == null) throw new ArgumentNullException(nameof(paddingSelector));
             if (width < 0) throw new ArgumentException(null, nameof(width));
-            return PadImpl(source, width, default, paddingSelector);
+            return PadImpl(source, width, default!, paddingSelector);
```

Comment:
Why? Shouldn't PadImpl accept a `T?`

Reply:
You can't express `T?` for `T`. See following passage in [Generic definitions and nullability](https://docs.microsoft.com/en-us/dotnet/csharp/nullable-migration-strategies#generic-definitions-and-nullability) section of the [nullable migration strategies](https://docs.microsoft.com/en-us/dotnet/csharp/nullable-migration-strategies documentation):

> What it does mean is that you can't use `T?` in a generic class or method declaration without constraints. For example, [`Enumerable.FirstOrDefault<TSource>(IEnumerable<TSource>)`](https://docs.microsoft.com/en-us/dotnet/api/system.linq.enumerable.firstordefault#System_Linq_Enumerable_FirstOrDefault__1_System_Collections_Generic_IEnumerable___0__) won't be changed to return `T?`.

Reply:
@atifaziz , You can add the ` [return:MaybeNull]` annotation to the method return using the  Nullable [Attributes](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/attributes/nullable-analysis#specify-post-conditions-maybenull-and-notnull) to express T?.

Reply:
@moh-hassan `MaybeNull` is a post-condition and we're not discussing the return type here.